spec/factories/users.rb

```diff
@@ -18,7 +18,11 @@
   factory :user do
     email                  { Faker::Internet.unique.email }
     password               { Faker::Internet.password(min_length: 8) } 
-    gender                 { Faker::Internet.gender } 
+    gender                 { Faker::Gender.binary_type } 
     uid                    { email }
+
+    factory :user_without_uid do
```

Comment:
Great job!!  In this case, you can call the factory and set the param in nil or '', something like  create(:user, uid: '') in the spec

Reply:
Thanks!